data/src/main/java/com/linkedin/data/schema/JsonBuilder.java

```diff
@@ -310,12 +337,12 @@ public void writeDataTemplate(DataTemplate<?> template, Boolean order) throws IO
     _jacksonDataTemplateCodec.dataTemplateToJsonGenerator(template, _jsonGenerator, order);
   }
 
-  private final StringWriter _writer;
+  private final Writer _writer;
   private final JsonGenerator _jsonGenerator;
-  private final JacksonDataCodec _jacksonDataCodec = new JacksonDataCodec();
-  private final JacksonDataTemplateCodec _jacksonDataTemplateCodec = new JacksonDataTemplateCodec();
 
-  private static final JsonFactory _jsonFactory = new JsonFactory().disable(JsonFactory.Feature.INTERN_FIELD_NAMES);
```

Comment:
@karthikbalasub do you know what is the intension to disable INTERNING in the first place?

Reply:
We disabled Jackson string interning to avoid OOM from PermGen space in 2013. String interning stopped using PermGen space starting from some intermediate update version of JDK 7. We should have removed this config long way back.

Reply:
In JacksonSmileStreamDataCodec, we have disable INTERNING, not sure if there is special reason? @karthikrg But in JacksonStreamDataCodec, we have started all using default value (enabled) and reuse JsonFactory based on https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance